src/librawspeed/decoders/ArwDecoder.h

```diff
@@ -23,36 +23,33 @@
 
 #include "common/Common.h"       // for uint32
 #include "common/RawImage.h"     // for RawImage
-#include "decoders/RawDecoder.h" // for RawDecoder, RawDecoderThread (ptr o...
+#include "decoders/TiffBasedDecoder.h"
 #include "io/FileMap.h"          // for FileMap
 
 namespace RawSpeed {
 
 class ByteStream;
 class CameraMetaData;
-class TiffIFD;
 
-class ArwDecoder :
-  public RawDecoder
+class ArwDecoder final : public TiffBasedDecoder
 {
 public:
-  ArwDecoder(TiffIFD *rootIFD, FileMap* file);
-  ~ArwDecoder() override;
+  using TiffBasedDecoder::TiffBasedDecoder;
```

Comment:
This sadly really breaks IWYU at the moment, https://github.com/include-what-you-use/include-what-you-use/issues/402 :((
Not that i want to insist on avoiding it